java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/ChangeMethodParametersRefactoring.java

```diff
@@ -0,0 +1,273 @@
+            path = path.getParentPath();
+        }
+        Element element = null;
+        FileObject elementSource = null;
+        if (kind == Tree.Kind.METHOD_INVOCATION || kind == Tree.Kind.NEW_CLASS || kind == Tree.Kind.MEMBER_REFERENCE) {
+            element = trees.getElement(path);
+            if (element == null || element.asType().getKind() == TypeKind.ERROR) {
+                return Collections.emptyList();
+            }
+            elementSource = SourceUtils.getFile(ElementHandle.create(element), info.getClasspathInfo());
+        }
+        if (elementSource == null) {
+            return Collections.emptyList();
+        }
+        QuickPickItem elementItem = new QuickPickItem(createLabel(info, element, true));
+        elementItem.setUserData(new ElementData(element));
+        return Collections.singletonList(createCodeAction(Bundle.DN_ChangeMethodParams(), CHANGE_METHOD_PARAMS_REFACTORING_KIND, null, CHANGE_METHOD_PARAMS_REFACTORING_COMMAND, Utils.toUri(elementSource), elementItem));
+    }
+
+    @Override
+    public Set<String> getCommands() {
+        return commands;
+    }
+
+    @Override
+    @NbBundle.Messages({
+        "DN_ChangeMethodSignature=Change method signature",
+    })
+    public CompletableFuture<Object> processCommand(NbCodeLanguageClient client, String command, List<Object> arguments) {
+        try {
+            if (arguments.size() > 1) {
+                String uri = gson.fromJson(gson.toJson(arguments.get(0)), String.class);
+                QuickPickItem sourceItem = gson.fromJson(gson.toJson(arguments.get(1)), QuickPickItem.class);
+                String label = sourceItem.getLabel();
+                int idx = label.indexOf('(');
+                client.showInputBox(new ShowInputBoxParams(Bundle.DN_ChangeMethodSignature(), label.substring(idx))).thenAccept(signature -> {
+                    if (signature != null && !signature.isEmpty()) {
+                        changeMethodParams(client, uri, sourceItem, signature);
+                    }
+                });
+            } else {
+                throw new IllegalArgumentException(String.format("Illegal number of arguments received for command: %s", command));
+            }
+        } catch (Exception ex) {
+            client.showMessage(new MessageParams(MessageType.Error, ex.getLocalizedMessage()));
+        }
+        return CompletableFuture.completedFuture(true);
+    }
+
+    private void changeMethodParams(NbCodeLanguageClient client, String uri, QuickPickItem source, String signature) {
+        try {
+            FileObject file = Utils.fromUri(uri);
+            ClasspathInfo info = ClasspathInfo.create(file);
+            JavaSource js = JavaSource.forFileObject(file);
+            if (js == null) {
+                throw new IOException("Cannot get JavaSource for: " + uri);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [IOException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [179].  Consider adding a catch clause of this specific exception in the outer try statement.